docs/source/contributing.rst

```diff
@@ -4,88 +4,139 @@ Contributing
-confusing.  Here is a short guide on how to do it.
+*  Code maintenance and development
+*  DevOps
+*  Documentation
+*  Developing educational content & narrative documentation
+*  Plugins to extend Manim functionality
+*  Testing (graphical, unit & video)
+*  Website design and development
 
+Contributing can be confusing, so here's a guide:
 
-Setup Manim and version control
+
+Manim Development Process
 ===============================
 
-#. *Make a fork of this repository on Github.*
-   You will need an account with Github. This will allow you to make pull requests (PR)
+For first-time contributors
+----------------------------
+#. Install git:
+
+   For instructions see https://git-scm.com/.
+
+
+#. Fork the project. Go to https://github.com/ManimCommunity/manim and
+   click the "fork" button to create your own copy of the project. You will
+   need a GitHub account. This will allow you to make a "pull request" (PR)
    to the ManimCommunity repo later on.
 
-#. *Clone your fork.*
-   You can clone your Github fork by running:
+#. Clone your fork to your local computer:
 
    .. code-block:: shell
 
-      git clone <your-fork-url>
-      cd manim
+      git clone https://github.com/<your-username>/manim.git
 
    GitHub will provide both a SSH (``git@github.com:<your-username>/manim.git``) and
-   HTTPS (``https://github.com/<your-username>/manim.git``) URL for cloning the repo.
-   You can use whichever one you are setup for.
+   HTTPS (``https://github.com/<your-username>/manim.git``) URL for cloning.
+   You can use SSH if you have SSH keys setup.
 
    .. WARNING::
 
-      Do not ``git clone`` the original ManimCommunity repository.  You must
-      clone your own fork.  After this step, there are three different
-      repositories to keep track of: the original ManimCommunity repo, your own
-      fork of it, and your local repository.
+      Do not clone the ManimCommunity repository. You must clone your own
+      fork.
+
+#.  Change the directory to enter the project folder:
+
+    .. code-block:: shell
+
+       cd manim
 
-#. Make ``git`` aware of the ManimCommunity repo.
+#. Add the upstream repository, ManimCommunity:
 
    .. code-block:: shell
 
       git remote add upstream https://github.com/ManimCommunity/manim.git
-      git fetch upstream
+ 
+#. Now, ``git remove -v`` should show two remote repositores named:
+   
+   - ``origin``, your forked repository
+   - ``upstream`` the ManimCommunity repository
 
-   After these commands, your local repository can keep track of your fork
-   (referred to as 'origin') as well as the main ManimCommunity repository
-   (referred to as 'upstream').
+#. Install Manim:
+ 
+   - See our :doc:`instructions for developers<installation/for_dev>` for
+     details and continue here afterwards
 
-#. *Install manim.*
-   See the :doc:`installation instructions for developers <installation/for_dev>` for
-   details.
+#. Install Pre-Commit (recommended):
```

Comment:
I would say it to be recommended. It irritates quite some times. but still, it should be fine I think

Reply:
to be not recommended?*
I'm mainly recommending it so that new contributors don't forget to run black